tests/api/test_models.py

```diff
@@ -80,10 +81,33 @@
 )
 def test_network_or_ip_parse(inp: str, mode: IPNetMode, expect: Any) -> None:
     """Test the network or IP address from string."""
-    res = NetworkOrIP.parse(inp, mode)
+    res = NetworkOrIP.parse_or_raise(inp, mode)
     assert res == expect
 
 
+@pytest.mark.parametrize(
+    "inp, expect_type",
```

Comment:
No biggie, but I'd prefer this to be `expect_type_call` to point out that this parameters ia a callable and not a type.